include/PathUtilities.h

```diff
@@ -0,0 +1,289 @@
+    Boole MEZZ_LIB IsPathAbsolute_Posix(const StringView ToCheck);
+    /// @brief Gets whether or not a path is absolute on Windows.
+    /// @details A path is absolute if it defines an explicit location of a resource.
+    /// @param ToCheck The path to check.
+    /// @return Returns true if the specified path is considered absolute on a Windows platform, false otherwise.
+    Boole MEZZ_LIB IsPathAbsolute_Windows(const StringView ToCheck);
+    /// @brief Gets whether or not a path is relative.
+    /// @warning This method is imperfect and doesn't check for a faulty path, just that it's not absolute.
+    /// @note A path is relative if it requires using the current working directory to define the location of
+    /// a resource.
+    /// @details This will check if the path is absolute on either Windows or Posix platforms.
+    /// @param ToCheck The path to check.
+    /// @return Returns true if the specified path is relative, false otherwise.
+    Boole MEZZ_LIB IsPathRelative(const StringView ToCheck);
+    /// @brief Gets whether or not a path is relative on the host.
+    /// @warning This method is imperfect and doesn't check for a faulty path, just that it's not absolute.
+    /// @note A path is relative if it requires using the current working directory to define the location of
+    /// a resource.
+    /// @details This will call the version of this method that matches the compiled platform.
+    /// @param ToCheck The path to check.
+    /// @return Returns true if the specified path is considered relative on the host platform, false otherwise.
+    Boole MEZZ_LIB IsPathRelative_Host(const StringView ToCheck);
+    /// @brief Gets whether or not a path is relative on Posix.
+    /// @warning This method is imperfect and doesn't check for some faults that can occur in a path, but does
+    /// take the extra time to verify the absence of Windows style directory separators (\\).
+    /// @note A path is relative if it requires using the current working directory to define the location of
+    /// a resource.
+    /// @param ToCheck The path to check.
+    /// @return Returns true if the specified path is considered relative on a Posix platform, false otherwise.
+    Boole MEZZ_LIB IsPathRelative_Posix(const StringView ToCheck);
+    /// @brief Gets whether or not a path is relative on Windows.
+    /// @warning This method is imperfect and doesn't check for some faults that can occur in a path, nor does
+    /// it check for proper Windows style directory separators as most uses of filesystem paths in the Windows
+    /// API can tolerate Windows or Posix syle.
+    /// @note A path is relative if it requires using the current working directory to define the location of
+    /// a resource.
+    /// @param ToCheck The path to check.
+    /// @return Returns true if the specified path is considered relative on a Windows platform, false otherwise.
+    Boole MEZZ_LIB IsPathRelative_Windows(const StringView ToCheck);
+
```

Comment:
Should there be a pair of functions IsPosixPath and IsWindowsPath? Some strings might be both, like relative paths, but anything starting with a drive letter and a colon is clearly windows while anything starting with a root slash is clearly posix. These could also encapsulate rules around prohibited characters (which might deserve their own functions too).

Reply:
Logic like you are describing already exists in the absolute path checks for Posix and Windows.